examples/paddle/mnist/test_single_gpu_mnist.py

```diff
@@ -0,0 +1,198 @@
+    parser = argparse.ArgumentParser(description="Paddle MNIST Example")
+    parser.add_argument(
+        "--batch-size",
+        type=int,
+        default=64,
+        metavar="N",
+        help="input batch size for training (default: 64)",
+    )
+    parser.add_argument(
+        "--test-batch-size",
+        type=int,
+        default=1000,
+        metavar="N",
+        help="input batch size for testing (default: 1000)",
+    )
+    parser.add_argument(
+        "--epochs",
+        type=int,
+        default=14,
+        metavar="N",
+        help="number of epochs to train (default: 14)",
+    )
+    parser.add_argument(
+        "--lr",
+        type=float,
+        default=0.001,
+        metavar="LR",
+        help="learning rate (default: 0.001)",
+    )
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        default=False,
+        help="quickly check a single pass",
+    )
+    parser.add_argument("--seed", type=int, default=1, metavar="S", help="random seed (default: 1)")
+    parser.add_argument(
+        "--log-interval",
+        type=int,
+        default=10,
+        metavar="N",
+        help="how many batches to wait before logging training status",
+    )
+    parser.add_argument("--use-te",
+                        action="store_true",
+                        default=False,
+                        help="Use Transformer Engine")
+    return parser.parse_args(args)
+
+
+def train_and_evaluate(args):
+    """Execute model training and evaluation loop."""
+    print(args)
+
+    paddle.seed(args.seed)
+
+    # Load MNIST dataset
+    transform = Normalize(mean=[127.5], std=[127.5], data_format='CHW')
+    train_dataset = MNIST(mode='train', transform=transform)
+    val_dataset = MNIST(mode='test', transform=transform)
+
+    # Define data loaders
+    train_loader = DataLoader(train_dataset, batch_size=args.batch_size, shuffle=True)
+    val_loader = DataLoader(val_dataset, batch_size=args.test_batch_size)
+
+    # Define model and optimizer
+    model = Net(use_te=args.use_te)
+    optimizer = paddle.optimizer.Adam(learning_rate=args.lr, parameters=model.parameters())
+
+    # Cast model to BF16
+    model = paddle.amp.decorate(models=model, level='O2', dtype='bfloat16')
+
+    for epoch in range(1, args.epochs + 1):
+        loss = train(args, model, train_loader, optimizer, epoch)
+        acc = evaluate(model, val_loader, epoch)
+
+    return loss, acc
+
+
+class TestMNIST(unittest.TestCase):
+    """MNIST unittests"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run MNIST without Transformer Engine"""
+        cls.args = mnist_parser(["--epochs", "5"])
+
+    @staticmethod
+    def verify(actual):
+        """Check If loss and accuracy match target"""
+        desired_traing_loss = 0.5
+        desired_test_accuracy = 0.98
+        assert actual[0] < desired_traing_loss
+        assert actual[1] > desired_test_accuracy
+
+    def test_te_bf16(self):
```

Comment:
I see a test failure on a V100:
```suggestion
    @pytest.mark.skipif(paddle.device.cuda.get_device_capability() < (8, 0),
        reason="BF16 MNIST test requires Ampere+ GPU")
    def test_te_bf16(self):
```

Reply:
Done.